sideboard/lib/sa/__init__.py

```diff
@@ -143,6 +143,43 @@ def process_result_value(self, value, engine):
     __all__.append('UTCDateTime')
 
 
+def check_constraint_naming_convention(constraint, table):
+    """Creates a unique name for an unnamed CheckConstraint.
+
+    The generated name is the SQL text of the CheckConstraint with
+    non-alphanumeric, non-underscore operators converted to text, and all
+    other non-alphanumeric, non-underscore substrings replaced by underscores.
+
+    If the generated name is longer than 32 characters, a uuid5 based on the
+    generated name will be returned instead.
+
+    >>> check_constraint_naming_convention(CheckConstraint('failed_logins > 3'), Table('account', MetaData()))
+    'failed_logins_gt_3'
+
+    See: http://docs.sqlalchemy.org/en/latest/core/constraints.html#configuring-constraint-naming-conventions
+    """
+    # The text of the replacements doesn't matter, so long as it's unique
+    replacements = [
```

Comment:
Out of curiosity, how did you come up with this list?  Presumably it's an exhaustive list; was it drawn from somewhere in the particular?